doc/mgmt/gnmi/SONiC_GNMI_Server_Interface_Design.md

```diff
@@ -0,0 +1,530 @@
+
+<img src="images/heartbeat.png" alt="heartbeat" width="500px"/>
+
+#### 1.2.1.8 Authentication
+sonic-telemetry provides three authentication mechanisms, and sonic-gnmi will use the same mechanisms:
+* Password: Like HTTP Basic auth, you pass the username and password in the gRPC metadata
+* JWT: JSON Web Tokens standard. First you authenticate with username/password and then receive a JWT token. After you send the token in the gRPC metadata.
+* User Certificate: Use a valid client certificate with the username embedded in the CN field. The certificate is for authentication, the username is for authorization.
+
+sonic-restapi uses one authentication mechanism:
+* In the authentication of client certificate, after the certificate chain is validated by TLS, it will further check if the common name of the end-entity certificate is in the trusted common name list of the server config.
+
+#### 1.2.1.9 ACL
+GNMI server will not support OpenConfig Yang models, so ACL configuration will use CONFIG_DB schema and SONiC Yang Models.
+
+#### 1.2.1.10 Docker to Host communication
+'config apply-patch' and 'config reload' are designed to run on host, and it's difficult to support them in container:
+1. These commands will update redis database and restart container, when they restart gnmi, bgp, syncd and swss, the ongoing gnmi operation will be broken.
+2. 'config reload' will stop service at first, run some other operations, and then restart service. If we run this command in container, it will be broken at the stop service step.
+3. These commands will execute some host scripts and use systemctl to restart service, it will be dangerous to support these operations in container.
+The solution is to add host services for 'config apply-patch' and 'config reload' on host, and gNMI server uses dbus method to invoke these services to update configuration
+
+##### 1.2.1.10.1 Incremental Configurations
+
+For incremental configuration, 'config apply-patch' should not restart gNMI, bgp, synd and swss.
+
+<img src="images/incremental_rpc.png" alt="incremental" width="500px"/>
+
+##### 1.2.1.10.2 Full Configurations
+
+For full configuration, there’re 2 possible solutions: 
+* gNMI server needs to send response at first, and then invoke 'config reload'.
+
+<img src="images/full_rpc.png" alt="full-gnmi" width="500px"/>
+
+* Use gNMI request and gNOI request together to implement full configuration update.
+
+<img src="images/full_rpc_gnoi.png" alt="full-gnmi-gnoi" width="500px"/>
+
+The full configuration request will be overwritten by subsequent full configuration request or incremental configuration request.
+
+<img src="images/mixed requests.png" alt="overwritten-config" width="500px"/>
+
+### 1.2.2 Container
+
+All the introduced features will be part of the sonic-gnmi package installed in sonic-gnmi container.
```

Comment:
There is no need for a new GNMI server. There is already one GNMI container which runs the server. Please enhance the existing container instead of creating a new one.